src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +266,237 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                {
+                    if (sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(linkingEntityName, out DatabaseObject? linkingDbObject))
+                    {
+                        ObjectTypeDefinitionNode node = SchemaConverter.GenerateObjectTypeDefinitionForDatabaseObject(
+                                entityName: linkingEntityName,
+                                databaseObject: linkingDbObject,
+                                configEntity: linkingEntity,
+                                entities: new(new Dictionary<string, Entity>()),
+                                rolesAllowedForEntity: new List<string>(),
+                                rolesAllowedForFields: new Dictionary<string, IEnumerable<string>>()
+                            );
+
+                        linkingObjectTypes.Add(linkingEntityName, node);
+                    }
+                }
+            }
+
+            return linkingObjectTypes;
+        }
+
+        /// <summary>
+        /// Helper method to generate object types for linking nodes from (source, target) using
+        /// simple linking nodes which represent a linking table linking the source and target tables which have an M:N relationship between them.
+        /// A 'sourceTargetLinkingNode' will contain:
+        /// 1. All the fields (column/relationship) from the target node,
+        /// 2. Column fields from the linking node which are not part of the Foreign key constraint (or relationship fields when the relationship
+        /// is defined in the config).
+        /// </summary>
+        /// <param name="objectTypes">Collection of object types.</param>
+        /// <param name="linkingObjectTypes">Collection of object types for linking entities.</param>
+        private void GenerateSourceTargetLinkingObjectDefinitions(
+            Dictionary<string, ObjectTypeDefinitionNode> objectTypes,
+            Dictionary<string, ObjectTypeDefinitionNode> linkingObjectTypes)
+        {
+            foreach ((string linkingEntityName, ObjectTypeDefinitionNode linkingObjectDefinition) in linkingObjectTypes)
+            {
+                (string sourceEntityName, string targetEntityName) = GraphQLUtils.GetSourceAndTargetEntityNameFromLinkingEntityName(linkingEntityName);
+                string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(targetEntityName);
+                ISqlMetadataProvider sqlMetadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                if (sqlMetadataProvider.GetEntityNamesAndDbObjects().TryGetValue(sourceEntityName, out DatabaseObject? sourceDbo))
+                {
+                    IEnumerable<ForeignKeyDefinition> foreignKeyDefinitionsFromSourceToTarget = sourceDbo.SourceDefinition.SourceEntityRelationshipMap[sourceEntityName].TargetEntityToFkDefinitionMap[targetEntityName];
+
+                    // Get list of all referencing columns in the linking entity.
```

Comment:
In case of M:N relationships, all the referencing columns belong to the linking entity. I'll update the comment and the variable name to be more clear.